### PR TITLE
[ME] Add manifest config `Hidden`; Add manifest template

### PR DESCRIPTION
**Pull request by acezen3d**
This is mainly to explicitly specify default values while keeping the properties hidden from the user.

### CODE REVIEW

File: src/MaterialEditor.Base/PluginBase.cs [hunk 1]
```diff
@@ -290,15 +291,17 @@ public class ShaderPropertyData
...
             public float? MaxValue;
 
-            public ShaderPropertyData(string name, ShaderPropertyType type, string defaultValue = null, string defaultValueAB = null, string minValue = null, string maxValue = null)
+            public ShaderPropertyData(string name, ShaderPropertyType type, string defaultValue = null, string defaultValueAB = null, string hidden = null, string minValue = null, string maxValue = null)
             {
                 Name = name;
                 Type = type;
                 DefaultValue = defaultValue.IsNullOrEmpty() ? null : defaultValue;
                 DefaultValueAssetBundle = defaultValueAB.IsNullOrEmpty() ? null : defaultValueAB;
+                Hidden = bool.TryParse(hidden, out bool result);
```
Comment (ManlyMarco) [inline src/MaterialEditor.Base/PluginBase.cs:304]: This seems to be a mistake, the return value only shows if parsing was successful, not if the property is actually set to true.

Comment (ManlyMarco) [reply src/MaterialEditor.Base/PluginBase.cs:304]: Maybe something like
```suggestion
                Hidden = bool.TryParse(hidden, out bool result) && result;
```

Comment (mog1347) [reply src/MaterialEditor.Base/PluginBase.cs:304]: Yeah, this is a mistake.

**Comment by acezen3d** [comment]
Tested on KK&KKS, good to go.

**Review by ManlyMarco — APPROVED** [review]

**Pull request merged**